aws/logs_monitoring/lambda_function.py

```diff
@@ -652,7 +675,35 @@ def awslogs_handler(event, context, metadata):
                 metadata[DD_HOST] = arn
 
     # Create and send structured logs to Datadog
-    for log in logs["logEvents"]:
+    # if multiline regex, iterate over log events and concatenate multiline logs into parent log
+    # child multiline logs must have matching timestamp and not match regex start pattern
```

Comment:
`must have matching` is it a hard guarantee on AWS side ? If yes, can you link the information in the comment please. If not I think it's a strong assumption we are making and I am worried that there could be edge cases where multiline logs have 1 digit different and more.

Reply:
that's a good question and feedback. 

This is just what I saw in sample logs I generated with a lambda. Spent some time investigating but haven't been able to confirm. I don't think this is documented as a hard guarantee anywhere. 

The matching timestamps requirement was meant to be a conservative approach so that any unrelated logs without the start pattern would not accidentally get appended to the previous log. My thinking here is that the new behavior, in the case of a multiline log with different timestamps on "child" lines, would be that the multiline log would not include the full log. These log lines of the multiline log wouldn't get lost, but just sent as their own individual lines, which is the same as the current behavior.

It's hard for me to be super confident here about these sort of edge cases without more thorough logs to test against from different aws services. do you have any advice here? I can try to spend some time collecting different logs from the different aws services if we feel like it will add confidence to this PR.